server/src/main/java/io/crate/execution/MultiPhaseExecutor.java

```diff
@@ -44,11 +45,13 @@ private MultiPhaseExecutor() {
     public static CompletableFuture<SubQueryResults> execute(Map<LogicalPlan, SelectSymbol> dependencies,
                                                              DependencyCarrier executor,
                                                              PlannerContext plannerContext,
-                                                             Row params) {
+                                                             Row params,
+                                                             BiFunction<LogicalPlan, PlannerContext, LogicalPlan> optimizer) {
```

Comment:
```suggestion
                                                             BiFunction<LogicalPlan, PlannerContext, LogicalPlan> optimize) {
```

It's a function to apply, not a entity.